### DISCUSSION TITLE
how to install pyg2.2

### DISCUSSION THREAD

**Discussion by MstarLioning** [pyg-team/pytorch_geometric#8329] [Q&A] [answered]
please!!I searched all over the internet and couldn't find any relevant information.

**Comment by xnuohz** [comment]
Maybe you can try to build it from source:)
https://github.com/pyg-team/pytorch_geometric/releases/tag/2.2.0

**Reply by MstarLioning** [reply]
Thankyou，Could you give me some advice

**Comment by xnuohz** [comment]
BTW, why not install the latest version, like 2.4.0?

**Reply by MstarLioning** [reply]
Hello, I am pleased to see your response. When replicating a paper, the author mentioned the following information:
`conda create -n graphgps python=3.10
conda activate graphgps

conda install pytorch=1.13 torchvision torchaudio pytorch-cuda=11.7 -c pytorch -c nvidia
conda install pyg=2.2 -c pyg -c conda-forge
pip install pyg-lib -f https://data.pyg.org/whl/torch-1.13.0+cu117.html

# RDKit is required for OGB-LSC PCQM4Mv2 and datasets derived from it.  
conda install openbabel fsspec rdkit -c conda-forge

pip install pytorch-lightning yacs torchmetrics
pip install performer-pytorch
pip install tensorboardX
pip install ogb
pip install wandb

conda clean --all`
However, I did not achieve the expected results using the author's original code, as there was an error in one part of the file at:
`from torch_geometric.graphgym.config import (cfg, dump_cfg,
                                             set_agg_dir, load_cfg,
                                             makedirs_rm_exist)`
                                             
It indicates that the reference 'set_agg_dir' cannot be found in 'config.py'. I have tried almost all versions of solutions but can't resolve this issue. Have you encountered a similar problem, or could you offer any advice? I would be very grateful!

**Reply by rusty1s** [reply]
You need to use PyG 2.0 for `set_agg_dir`.

**Reply by MstarLioning** [reply]
@rusty1s Thank you for your response, but I also want to invoke some commands from section 2.2. Is there a way for me to enable the set_agg_dir function in section 2.2?

**Reply by MstarLioning** [reply]
@rusty1s And the author specifically states in the given environment that the Python version is 3.1.0, but I see that pyg2.0 requires a version less than 3.1.0. How should this be resolved? (https://github.com/rampasek/GraphGPS)Thank you!

**Reply by xnuohz** [reply, marked as answer]
If you just want to understand and run through the `GraphGPS` model for now, I think this example can help you
https://github.com/pyg-team/pytorch_geometric/blob/master/examples/graph_gps.py

**Reply by MstarLioning** [reply]
@xnuohz Thank you for your suggestions. As an undergraduate who has just started learning about graph machine learning, I feel there is a lot for me to understand slowly. However, the reason I ask is that I am participating in a Kaggle competition where there is a method called GST (https://github.com/kaidic/GST). This model borrows the environment from GST, but since I have little experience, I have not managed to properly integrate the environment so far. Regardless, thank you for your kindness, I will continue to work hard!
